ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Block/SupportsPeras.hs

```diff
@@ -83,6 +139,39 @@ data ValidatedPerasCert blk = ValidatedPerasCert
   deriving stock (Show, Eq, Ord, Generic)
   deriving anyclass NoThunks
 
+instance
+  ( HasId (PerasCert blk)
+  , Ord (IdOf (PerasCert blk))
+  , Eq (IdOf (PerasCert blk))
+  , Show (IdOf (PerasCert blk))
+  , NoThunks (IdOf (PerasCert blk))
+  , Serialise (IdOf (PerasCert blk))
```

Comment:
Out of curiosity, do you need all those instances when you already know `HasId (PerasCert blk)` and `type IdOf (ValidatedPerasCert blk) = IdOf (PerasCert blk)`?
I'm not saying you should change anything, but it seems redundant to me. Maybe you could say something like `IdOf (PerasCert blk) ~ IdOf (ValidatedPerasCert blk)` in the constraints?

Reply:
Hum HLS was complaining when I only had `HasId (PerasCert blk) =>` and suggested that I introduced the apparently-redundant constraints.

I'll try with `(HasId (PerasCert blk), IdOf (PerasCert blk) ~ IdOf (ValidatedPerasCert blk)) =>`! This may be the part that I was missing.